rclcpp/test/rclcpp/test_executor.cpp

```diff
@@ -488,6 +517,37 @@ TEST_F(TestExecutor, spin_until_future_complete_future_already_complete) {
     dummy.spin_until_future_complete(future, std::chrono::milliseconds(1)));
 }
 
+TEST_F(TestExecutor, spin_until_complete_condition_already_complete) {
+  DummyExecutor dummy;
+  auto condition = []() {return true;};
+  EXPECT_EQ(
+    rclcpp::FutureReturnCode::SUCCESS,
+    dummy.spin_until_complete(condition, std::chrono::milliseconds(1)));
+}
+
+TEST_F(TestExecutor, spin_until_complete_returns_after_condition) {
+  DummyExecutor dummy;
+  auto node = std::make_shared<rclcpp::Node>("node", "ns");
+  bool spin_called = false;
```

Comment:
nitpick: isn't this more like?

```suggestion
  bool timer_called = false;
```

Reply:
makes sense, renamed